### ISSUE TITLE
Expect to throw a `TypeError` got failed, `actual: TypeError, expected: TypeError`

### ISSUE THREAD

**Issue by feryardiant** [kahlan/kahlan#384]
When I'm trying to test a method to throw `TypeError`, I got the following message

```sh
Projek\Container
  Projek\Container::set
    ✖ it should throw TypeError when trying to depends on invalid entries
      expect->toThrow() failed in `.test\spec\Container.spec.php` line 212

      It expect actual to throw a compatible exception.

      actual:
        (object) an instance of `TypeError`
      expected:
        (object) an instance of `TypeError`
```

The error only happened when I passed the `Error` message.

```php
expect(...)->toThrow(new TypeError('Argument #1 ($cb) must be of type SomeClass, OtherBar given'));
```

But when I don't pass any error message like the following, the test passed

```php
expect(...)->toThrow(new TypeError);
```

So I tried to run the test and `expect()` nothing, just to see what's the actual error message and I got this

```sh
Projek\Container
  Projek\Container::set
    ✖ it should throw TypeError when trying to depends on invalid entries
      an uncaught exception has been thrown in `test\spec\Container.spec.php` line 204

      message:`TypeError` Code(0) with message "Kahlan\\Cli\\Kahlan::{closure}(): Argument #1 (\$cb) must be of type SomeClass, OtherBar  given"

        [NA] - test\spec\Container.spec.php, line 197 to 204
        Closure::__invoke() - [internal], line ?
        ReflectionMethod::invokeArgs() - src\Container\Resolver.php, line 48
        Projek\Container\Resolver::handle() - src\Container.php, line 80
        Projek\Container::get() - test\spec\Container.spec.php, line 209
        // ...
```
Then I try to run the test with the following error message.

```php
expect(...)->toThrow(new TypeError('Kahlan\Cli\Kahlan::{closure}(): Argument #1 ($cb) must be of type SomeClass, OtherBar given'));
```
And the test is passed.

**Comment by jails** [comment]
When you define a specific error message to an Exception it must matches the throwed one.
What make you feel there's an issue here ?

**Comment by feryardiant** [comment]
> it must matches the throwed one
@jails exactly! but, as we can see the error message was

```
message:`TypeError` Code(0) with message "Kahlan\\Cli\\Kahlan::{closure}(): ...."
```
It seems like it was thrown from a closure inside `Kahlan\Cli\Kahlan` namespace instead of the test file, which doesn't have namespace.

Is it actually expected behavior?

**Comment by feryardiant** [comment]
Also the `failed message` doesn't show the complete error message, or I could say it less informative.
```
actual:
  (object) an instance of `TypeError`
expected:
  (object) an instance of `TypeError`
```
To me, it sounds like _"Hey, go have a test failed. You expecting an instance of TypeError, but you got an instance of TypeError"_. What's make it failed?

I hope it could be more informative, or at least it should behave like any other tests using `toThrow()`. e.g. I have a test as follow

```php
expect(function () {
    // ...
})->toThrow(new SomeException('Some expected message'));
```
I'd expect the test should give me something like this if the test was failed.
```sh
It expect actual to throw a compatible exception.

actual:
  (object) an instance of `SomeException` Code(0) with message "Some actual message"
expected:
  (object) an instance of `SomeException` Code(0) with message "Some expected message"
```

So I can easily understand what's missing and what's makes my test failed,

**Comment by jails** [comment]
> Is it actually expected behavior?

Well, as you can see, PHP prefixes error messages with the "callable context" for throwed TypeError, so "Kahlan\Cli\Kahlan::{closure}():" is something added by PHP itself.

```
<?php
namespace Simple\Test;
use DateTime;
use Throwable;

function test(DateTime $datetime) {}

try {
    test(null);
} catch (Throwable $e) {
    echo $e->getMessage(); // Simple\Test\test(): Argument #1 ($datetime) must be of type DateTime, null given, called in [...][...] on line 11
}
```

> I'd expect the test should give me something like this if the test was failed.

Indeed, I fixed it in the 5.1.1 release.
